lib/omniauth/strategies/openid_connect.rb

```diff
@@ -201,7 +201,18 @@ def discover!
       end
 
       def user_info
-        @user_info ||= access_token.userinfo!
+        @user_info ||=
+          begin
+            userinfo = access_token.userinfo!
+
+            if access_token.id_token
+              decoded = decode_id_token(access_token.id_token).raw_attributes
+
+              ::OpenIDConnect::ResponseObject::UserInfo.new userinfo.raw_attributes.merge(decoded)
+            else
+              userinfo
```

Comment:
Please consider changing to something more compact:

```ruby
def user_info
  return @user_info if @user_info

  if access_token.id_token
    decoded = decode_id_token(access_token.id_token).raw_attributes
    @user_info = ::OpenIDConnect::ResponseObject::UserInfo.new(userinfo.raw_attributes.merge(decoded))
  else
    @user_info = access_token.userinfo!
  end
end
```

Reply:
thanks, updated! :+1: